demo/react/doc/product/components/autocomplete.mdx

```diff
@@ -0,0 +1,130 @@
+```javascript import
+import { Autocomplete, List, ListItem, ListItemSize, Size, TextField, Chip, ListDivider, Icon, ListSubheader } from 'LumX';
+import { useBooleanState } from 'LumX/core/react/hooks';
+import { DOWN_KEY_CODE, ENTER_KEY_CODE, TAB_KEY_CODE, UP_KEY_CODE } from 'LumX/core/constants';
+```
+
+# Autocomplete
+
+## Default
+
+```javascript jsx withThemeSwitcher
+(theme) => {
+    const CITIES = [
+        {
+            text: 'Los Angeles',
+            id: 'losangeles'
+        },
+        {
+            text: 'San Francisco',
+            id: 'sanfrancisco'
+        },
+        {
+            text: 'Paris',
+            id: 'paris'
+        },
+        {
+            text: 'Montpellier',
+            id: 'montpellier'
+        },
+        {
+            text: 'Bordeaux',
+            id: 'bordeaux'
+        },
+        {
+            text: 'Toulouse',
+            id: 'toulouse'
+        },
+        {
+            text: 'Lyon',
+            id: 'lyon'
+        },
+        {
+            text: 'Montevideo',
+            id: 'montevideo'
+        }
+    ];
+
+    const [showSuggestions, setShowSuggestions] = React.useState(false);
+    const [activeIndex, setActiveIndex] = React.useState(-1);
```

Comment:
I think it could be possible to handle the active element index in a custom hook that we could use both here and in the List component.

We might have to refactor the List/ListItem components to extract there logic

We could do something like this: 
```jsx
    // Listen to keydown on textField to handle keyboard selection.
    const [activeIndex, onSelectedItem] = useKeyboardSelection(textFieldRef);
    onSelectedItem(selectedIndex => /* do something */); 

    return (
        <Autocomplete ... >
            <List isClickable>
                {filteredCities.map((city, index) => (
                        <ListItem isActive={index === activeIndex}>
                            <div>{city.text}</div>
                        </ListItem>
```

Reply:
We decided to go for the custom hook but for now use it on the Autocomplete component, in order to have a lower impact on the product. later on, it would be great to have that hook implemented on the List as well.

Reply:
@gcornut I uploaded the first version of the custom hook!